ykrt/src/compile/jitc_yk/codegen/x64/deopt.rs

```diff
@@ -14,6 +14,17 @@ use yksmp::Location as SMLocation;
 
 use super::{X64CompiledTrace, RBP_DWARF_NUM, REG64_SIZE};
 
+/// Registers (in DWARF notation) that we want to restore during deopt. Excludes `rsp` (7) and
+/// `return register` (16), which we do not care about.
+const RECOVER_REG: [usize; 31] = [
+    0, 1, 2, 3, 4, 5, 6, 8, 9, 10, 11, 12, 13, 14, 15, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27,
+    28, 29, 30, 31, 32,
+];
+
+/// Number of normal + float registers including the ones missing above. Used to initialise
+/// the register array storing live register values during deopt.
+const REGISTER_NUM: usize = RECOVER_REG.len() + 2;
```

Comment:
Remind me why this ends up being 33?

Reply:
33 (dwarf) registers covers all of the GP and XMM registers.

![image](https://github.com/user-attachments/assets/23f7a013-bc68-4d6b-a147-cc7782a5668a)

The +2 is because `RECOVER_REG` omits two registers that we shouldn't interfere with during deopt (but we still wish to use the numbers as array indices later).

Hard to explain. Hope it makes sense?

Reply:
Let's at least briefly document that: it's weird and unexpected.

Reply:
I think that's what the docstring says?

Reply:
No, it doesn't explain why it's 33 (which is a very weird number in this context!), and it doesn't mention DWARF or RECOVER_REG or whatnot.